test/jb_nft_reward_datasource_delegate/did_pay.test.js

```diff
@@ -0,0 +1,191 @@
+
+    await expect(jbNFTRewardDataSource.connect(projectTerminal).didPay({
+      payer: beneficiary.address,
+      projectId: PROJECT_ID,
+      currentFundingCycleConfiguration: 0,
+      amount: { token: ethToken, value: ETH_TO_PAY, decimals: 18, currency: CURRENCY_ETH },
+      projectTokenCount: 0,
+      beneficiary: beneficiary.address,
+      preferClaimedTokens: true,
+      memo: '',
+      metadata: '0x42'
+    })).to.emit(jbNFTRewardDataSource, 'Transfer').withArgs(ethers.constants.AddressZero, beneficiary.address, 0);
+  });
+
+  it(`Should not mint token if exceeding max supply`, async function () {
+    const { jbNFTRewardDataSource, projectTerminal, beneficiary } = await setup();
+
+    await jbNFTRewardDataSource.connect(projectTerminal).didPay({
+      payer: beneficiary.address,
+      projectId: PROJECT_ID,
+      currentFundingCycleConfiguration: 0,
+      amount: { token: ethToken, value: ETH_TO_PAY, decimals: 18, currency: CURRENCY_ETH },
+      projectTokenCount: 0,
+      beneficiary: beneficiary.address,
+      preferClaimedTokens: true,
+      memo: '',
+      metadata: '0x42'
+    });
+
+    await expect(jbNFTRewardDataSource.connect(projectTerminal).didPay({
+      payer: beneficiary.address,
+      projectId: PROJECT_ID,
+      currentFundingCycleConfiguration: 0,
+      amount: { token: ethToken, value: ETH_TO_PAY, decimals: 18, currency: CURRENCY_ETH },
+      projectTokenCount: 0,
+      beneficiary: beneficiary.address,
+      preferClaimedTokens: true,
+      memo: '',
+      metadata: '0x42'
+    })).not.to.emit(jbNFTRewardDataSource, 'Transfer');
+  });
+
+  it(`Should not mint token if contribution below limit`, async function () {
+    const { jbNFTRewardDataSource, projectTerminal, beneficiary } = await setup();
+
+    await expect(jbNFTRewardDataSource.connect(projectTerminal).didPay({
+      payer: beneficiary.address,
+      projectId: PROJECT_ID,
+      currentFundingCycleConfiguration: 0,
+      amount: { token: ethToken, value: 0, decimals: 18, currency: CURRENCY_ETH },
+      projectTokenCount: 0,
+      beneficiary: beneficiary.address,
+      preferClaimedTokens: true,
+      memo: '',
+      metadata: '0x42'
+    })).not.to.emit(jbNFTRewardDataSource, 'Transfer');
+  });
+
+  it(`Should not mint token if not called from terminal`, async function () {
+    const { jbNFTRewardDataSource, beneficiary } = await setup();
+
+    await expect(jbNFTRewardDataSource.connect(beneficiary).didPay({
+      payer: beneficiary.address,
+      projectId: PROJECT_ID,
+      currentFundingCycleConfiguration: 0,
+      amount: { token: ethToken, value: 2000, decimals: 18, currency: CURRENCY_ETH },
+      projectTokenCount: 0,
+      beneficiary: beneficiary.address,
+      preferClaimedTokens: true,
+      memo: '',
+      metadata: '0x42'
+    })).to.be.revertedWith('INVALID_PAYMENT_EVENT()');
+
+  });
+
+  it(`Tests for unsupported pay functions`, async function () {
```

Comment:
Those are returning values (and are therefore 'supported' since they're pointing to the correct delegate to use later on in pay(..)) + returning a weight of 0 (ie 0 project token minted)